.dockerignore

```diff
@@ -0,0 +1,44 @@
+.cache
+circle.yml
+README.md
+
+**/*.class
+*.bak
+logs
+.project
+.classpath
+.settings
+gradle.properties
+.gradle
+*~
+myClock.bin
+/*.tar.gz
+*.class
+**/bin
+
+# ignore build directories
+**/build
+**/target
+**/devel
+
+# ignore Intellij project files and dirs
```

Comment:
Are we going to need to add checks like this for all our Ide's (Intellij, Eclipse, VSCode, etc.)

Reply:
We can, yeah, I just copied in our .gitignore as a base so if we have other editors that are worth adding here we can certainly do so.  Nominally one might have those in their personal global gitignore file but redundancy is good in this case. I think this can be grown as needed though, don't need to take care of that under the scope of this PR.